includes/Classifai/Providers/OpenAI/Embeddings.php

```diff
@@ -412,23 +437,29 @@ public function generate_embeddings_for_post( $post_id ) {
 
 		// Only run on supported post types and statuses.
 		if (
-			! in_array( $post->post_type, $this->supported_post_types(), true ) ||
-			! in_array( $post->post_status, $this->supported_post_statuses(), true )
+			! $dryrun
+			&& (
+				! in_array( $post->post_type, $this->supported_post_types(), true ) ||
+				! in_array( $post->post_status, $this->supported_post_statuses(), true )
+			)
 		) {
 			return;
 		}
 
 		// Don't run if turned off for this particular post.
-		if ( 'no' === get_post_meta( $post_id, '_classifai_process_content', true ) ) {
+		if ( 'no' === get_post_meta( $post_id, '_classifai_process_content', true ) && ! $dryrun ) {
 			return;
 		}
 
 		$embeddings = $this->generate_embeddings( $post_id, 'post' );
 
 		// Add terms to this item based on embedding data.
 		if ( $embeddings && ! is_wp_error( $embeddings ) ) {
-			update_post_meta( $post_id, 'classifai_openai_embeddings', array_map( 'sanitize_text_field', $embeddings ) );
-			$this->set_terms( $post_id, $embeddings );
+			if ( ! $dryrun ) {
+				update_post_meta( $post_id, 'classifai_openai_embeddings', array_map( 'sanitize_text_field', $embeddings ) );
+			}
+
+			return $this->set_terms( $post_id, $embeddings, $dryrun );
```

Comment:
It seems most of the code we are adding to the `set_terms` method is specific for doing a dry run. Wondering if instead of having a big conditional in that method, what if we add a new method that we call here if `$dryrun` is true?

Something like:
```php
if ( $dryrun ) {
    return $this->get_terms( $post_id, $embeddings );
} else {
	update_post_meta( $post_id, 'classifai_openai_embeddings', array_map( 'sanitize_text_field', $embeddings ) );
    return $this->set_terms( $post_id, $embeddings );
}
```